aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -840,40 +1288,249 @@ def __init__(
         self._cors_enabled: bool = cors is not None
         self._cors_methods: Set[str] = {"OPTIONS"}
         self._debug = self._has_debug(debug)
+        self._enable_validation = enable_validation
         self._strip_prefixes = strip_prefixes
         self.context: Dict = {}  # early init as customers might add context before event resolution
         self.processed_stack_frames = []
 
         # Allow for a custom serializer or a concise json serialization
         self._serializer = serializer or partial(json.dumps, separators=(",", ":"), cls=Encoder)
 
+        if self._enable_validation:
+            from aws_lambda_powertools.event_handler.middlewares.openapi_validation import OpenAPIValidationMiddleware
+
+            self.use([OpenAPIValidationMiddleware()])
+
+            # When using validation, we need to skip the serializer, as the middleware is doing it automatically.
+            # However, if the user is using a custom serializer, we need to abort.
+            if serializer:
+                raise ValueError("Cannot use a custom serializer when using validation")
+
+            # Install a dummy serializer
+            self._serializer = lambda args: args  # type: ignore
```

Comment:
If we don't do this, our serializer will kick in before the middleware has a chance to see the code.